src/lib/storage/chunk.hpp

```diff
@@ -151,6 +152,21 @@ class Chunk : private Noncopyable {
    */
   size_t estimate_memory_usage() const;
 
+  void increase_invalid_row_count(uint64_t count) const;
+
+  uint64_t invalid_row_count() const { return _invalid_row_count; }
+
+  /**
+   * The MvccDeletePlugin implements a logical chunk deletion (full chunk invalidation) as a transaction.
+   * In case a clean-up transaction has been performed successfully, its commit id will be returned via
+   * this function. Otherwise, an empty _cleanup_commit_id is returned.
+   * The _cleanup_commit_id is used to determine if the chunk got already fully invalidated by the logical
+   * delete and can be physically deleted.
+   */
```

Comment:
This comment does not appear to be really helpful for someone looking at the chunk interface. What is a clean-up transaction? Why are chunks "fully invalidated"? ...?

I was thinking about something like "Chunks with few visible entries can be periodically cleaned up by the MvccDeletePlugin. In one step, it deletes the rows from this chunk and re-inserts them add the end of the table. Thus, future transactions will find the still valid rows at the end of the table and do not have to look at this chunk. The cleanup commit ID is the highest snapshot commit ID for which there are still visible."

Reply:
Great, thank you for your detailed suggestion.